test/integration/IntegrationChecks.t.sol

```diff
@@ -0,0 +1,418 @@
+        for (uint i = 0; i < churnedOperators.length; i++) {
+            User churnedOperator = churnedOperators[i];
+
+            bytes memory churnedQuorum = new bytes(1);
+            churnedQuorum[0] = churnedQuorums[i];
+
+            // RegistryCoordinator
+            assert_HasOperatorInfoWithId(churnedOperator, 
+                "churned operatorInfo should still have operatorId");
+            assert_NotRegisteredForQuorums(churnedOperator, churnedQuorum,
+                "churned operator bitmap should not include churned quorums");
+            assert_Snap_Deregistered_FromQuorums(churnedOperator, churnedQuorum,
+                "churned operator did not deregister from churned quorum");
+
+            // BLSApkRegistry
+            assert_HasRegisteredPubkey(churnedOperator, 
+                "churned operator should still have a registered pubkey");
+
+            // StakeRegistry
+            assert_NoExistingStake(churnedOperator, churnedQuorum,
+                "operator should no longer have stake in any quorums");
+        }
+    }
+
+    /*******************************************************************************
+                                 BALANCE UPDATE CHECKS
+    *******************************************************************************/
+
+    function check_Deposit_State(
+        User operator,
+        bytes memory quorums,
+        IStrategy[] memory strategies,
+        uint[] memory tokenBalances
+    ) internal {
+        _log("check_Deposit_State", operator);
+
+        // RegistryCoordinator
+        assert_Snap_Unchanged_OperatorInfo(operator,
+            "operator info should not have changed");
+        assert_Snap_Unchanged_QuorumBitmap(operator,
+            "operators quorum bitmap should not have changed");
+        
+        // BLSApkRegistry
+        assert_Snap_Unchanged_QuorumApk(quorums,
+            "quorum apks should not have changed");
+
+        // StakeRegistry
+        assert_Snap_Increased_OperatorWeight(operator, quorums,
+            "operator weight should not have decreased after deposit");
+        assert_Snap_Unchanged_OperatorStake(operator, quorums, 
+            "operator stake should be unchanged");
+        assert_Snap_Unchanged_TotalStake(quorums, 
+            "total stake should be unchanged");
+
+        // IndexRegistry
+        assert_Snap_Unchanged_OperatorCount(quorums,
+            "operator counts should not have changed");
+        assert_Snap_Unchanged_OperatorListEntry(quorums,
+            "operator list should not have changed");
+
+        // Core
+        assert_Snap_Added_OperatorShares(operator, strategies, tokenBalances,
+            "operator should have additional stake");
+    }
+
+    /// @dev Checks that an operator's stake was successfully increased
+    /// NOTE: This method assumes (and checks) that the operator already
+    ///       met the minimum stake before stake was added.
+    function check_DepositUpdate_State(
```

Comment:
perhaps more name differentiation between this method and `check_Deposit_State` could be good?
I have yet to look through the usage but it seems like `check_Deposit_State` is meant to be used after a "deposit to core" and then `check_DepositUpdate_State` is meant to be used after a follow-up "stake update" is triggered on the AVS?
Perhaps `check_DepositCore_State` and `check_DepositStakeUpdate_State` or something might be clearer?

Reply:
or maybe just another comment could help. idk. things are clean, there's just a lot to keep straight!

Reply:
There are a handful of unclear names, so I appreciate feedback like this. Will do!